### ISSUE TITLE
Check box support

### ISSUE THREAD

**Issue by busimus** [lervag/wiki.vim#1]
I really like the check boxes in the [vimwiki](https://github.com/vimwiki/vimwiki) plugin:
```
- [o] Do these things:
    - [X] This thing is done
    - [ ] Have yet to do this thing
```
Any plans on implementing support for them in this plugin?

**Comment by lervag** [comment]
I assume you already know that a simple check boxes feature exists already, i.e. one can use `<c-s>` in normal mode to toggle between `[ ]` and `[x]`. However, there is no propagation in multilevel lists.

I do agree this is a nice feature, and I would not mind implementing it. However, I do feel like there are some more important tasks that must be done first, and so I don't want to promise that this will be solved in very near future. I'll add the issue to the TODO list on the README file, which will allow potential guest contributors a chance to notice and help implement it.

**Comment by Busimus** [comment]
I actually didn't know that. I tried `<c-space>`, like in vimwiki, and searched the code for "check"; both of these things had no result, so I assumed there was no support for them at all. But now I see that they are just a part of lists.
But `<c-s>` works well enough for now, thanks. 

Since they're a part of lists, then I guess it's worth mentioning here that auto-creation of the next list item when a user presses enter in a list is something I also miss from vimwiki. `<c-s>` helps with this, but it doesn't create a checkbox when you're in a list with checkboxes.

**Comment by lervag** [comment]
Thanks, I will look into auto creation as well. I've currently relied on the Vim implementation of lists, which I don't think will be able to handle this directly, so it will probably require that I implement a more general way of handling lists. I don't really want to make this very complicated, so it would be nice to have a short discussion if which features are the most "essential". I'll start:

* [ ] Toggling checkboxes will propogate through a list
* [ ] Auto create checkbox for new list entry (when current entry is a checkbox)

Also, I know vimwiki used `<c-space>`, but this key is not easy to use in terminal Vim, so I decided to change it to `<c-s>`. I'll make this configurable in "near" future.

**Comment by lervag** [comment]
I'm getting there. This is not complete yet, but there is now some support for a more connected list toggle feature. Currently there is only a boolean state (on/off), but I'll look at that next.

**Comment by lervag** [comment]
It's been quite some time since the last work on this, but I've now finally made some more progress. I've added `g:wiki_list_todo` to allow to add more todo toggles. The next step is to allow more checkbox toggles, as well as the above "auto create" task.

**Comment by lervag** [comment]
Thanks for commenting!

So, I looked into this now. If you go to `wiki.vim/autoload/wiki.vim` to line 28 and 29, you can get what you want by removing the `f`s in the `comments` option. However, to me, it does not seem to be optimal, because I often write list entries that span more than a single line. In this case, I found that the behaviour becomes unexpected. As such, I ended up instead adding the `<c-s>` mapping, which makes it easy to add the list header. However, this does not work for the checklists right now.

I see two options for this:

1. Change the `comments` option so that checklist entries are prefilled. This works well if you keep the list entries to a single line, but textwrapping will add a new bullet. I guess this is OK, you will need to manually fixup, but probably less then the current method.

2. Keep things as they are, but update `<c-s>` to work for checklists.

**Comment by lervag** [comment]
I've made the change for 1 to see how it works. Let me know what you think about it.

I'm closing the issue for now, but let the discussion continue.
